shell/server/core.js

```diff
@@ -147,6 +148,64 @@ class NotificationHandle {
   }
 }
 
+class PersistentUiViewImpl {
+  constructor(token, sturdyRef, membraneRequirements) {
+    this._token = token;
+    this._sturdyRef = sturdyRef;
+    // We need this token so we can later pass it to SandstormPermissions.createNewApiToken, which
+    // requires the sturdyRef of the token it's producing a child for, not just the hashed token id.
```

Comment:
Yeah, that sounds useful!

One thing to keep aware of is that because tokens can chain, we want to make sure we're creating new child tokens as children of the token passed to restore(), not some parent token of that token that gives the restore()'d token any authorization.  Indeed, `this._token` is actually possibly a parent token with `owner: grain`, as looked up by `restoreInternal()` - see the

```} else if (token.parentToken) {```

 branch.